drafts/build_a_dao_on_fuel/draft.md

```diff
@@ -0,0 +1,1095 @@
+
+**Fuel**: [Fuel](https://fuel.network/) is an operating system purpose-built for Ethereum rollups, designed to help developers build decentralized economies at scale....
+
+**FuelVM**: The [Fuel Virtual Machine](https://medium.com/blockchain-capital-blog/exploring-the-fuelvm-86cf9ccdc159) responsible for managing all code execution on the Fuel Blockchain
+
+**Sway**: [Sway](https://fuellabs.github.io/sway/) is a domain-specific language used for writing smart contracts on the fuel blockchain.
+
+**ABI**: These are called [Application Binary Interfaces](https://docs.fuel.network/guides/intro-to-sway/contract-abi/), they are the blueprint, or interaction models of your smart contract, it is a layout of the detailed functionalities in your smart contract, and how to interact with it.
+
+Now that settled, here's where the fun part starts, It's time to get your hands dirty with coding.
+
+## Setting up your Environment
+
+In the next few steps, you will set up your development environment to get started with building your smart contract and the front end of your Dao.
+
+1. Go to your terminal and run the command `wsl` to start your `wsl` terminal
+
+2. Next run the command to generate your new contract directory.
+
+```bash
+mkdir Fuel_Dao
+cd Fuel_Dao
+forc new contract
+```
+
+
+## Writing your Smart Contract
+
+1. Head to the `main.sw` with the command `cd main.sw` contract file, where you will be implementing your main contracts. Following the next few steps, you will be replacing the code therein.
+
+2. First you define the file as a contract file, and import other contract libraries and interfaces from the following `crates`.
+
+    - `error` - define your error handlers for your contract.
+    - `data_structre` - defines functionalities for managing proposals, voting, and defines methods for initializing your DAO.
+    - `events` - defines a library of multiple `structs` to **emit events** when certain instances, or functions are executed on-chain.
+
+    - `utils` - this will instances that will be emitted on-chain when certain events are met.
+    - `interface` - this is where you will implement you contract ABIs
+
+    All of which will require you to create other types of sway contracts, hold that thought 👍.
+
+    Copy and replace the existing code in your `main.sw` file with the one below.
+
+[note]: <> (I use Rust for all snippets rather than sway since both languages are similar and Rust's is recognized on prettifier for markdown code snippets)
+    
+```rust 
+    contract;
+
+    mod data_structures;
+    mod errors;
+    mod events;
+    mod interface;
+    mod utils;
+```
+
+3.  Next, you will import the following modules from the sway standard library; 
+    - `asset::transfer` - This provides the functionality for transferring asset
+    - `auth::mes_sender` - The module retrieves the address of the sender of the contract or a transaction.
```

Comment:
`auth::mes_sender` is a typo.

Reply:
Missed that too, Thanks for pointing out